src/gui/filedetails/sharemodel.cpp

```diff
@@ -80,6 +81,7 @@ QHash<int, QByteArray> ShareModel::roleNames() const
     roles[PasswordRole] = "password";
     roles[PasswordEnforcedRole] = "passwordEnforced";
     roles[EditingAllowedRole] = "editingAllowed";
+    roles[IsSecureFileDropLinkRole] = "isSecureFileDropLink";
```

Comment:
@mgallien Because we do not introduce a new type of shares. They all have the enum type `TypeLink`. If I introduce a new type of link, it will not simplify code but merely add much more changes as all those parts where `TypeLink` is assumed will stop working. Does it make sense?

Reply:
sure, I would been fine with the current role but as soon as we start adding more `is...` roles we will need to find another way to do this